### ISSUE TITLE
Sistema de Temporadas

### ISSUE THREAD

**Issue by DanielHe4rt** [he4rt/he4rt-bot-next#13]
O Sistema de Temporadas tem o propósito de trazer anualmente uma sensação de renovação ao servidor. Como nós sabemos, temos um sistema de **leveling** implementado no bot desde o primeiro dia porém as pessoas estão upando já desde o inicio da comunidade.

A ideia é salvar a posição da pessoa na temporada passada e mostrar isso no seu perfil.
<p align="center">
<img src="https://user-images.githubusercontent.com/6912596/206874677-48ba5345-df2a-487a-a818-d7663d9aeffb.png" />
<i> Modelo do perfil do usuário pra temporada de 2023 </i>
</p>

A temporada atual durou 4 anos e irá finalizar dia 31 de dezembro de 2022. A partir de 01/01/2023, todos os usuários voltarão pro nível 1 com 0 de experiência e poderão reconquistar o topo do ranking, que hoje, pertence aos membros que ficaram esses 4 anos dentro da nossa comunidade.

Os dados referentes a quantidade de mensagens enviadas, eventos, reuniões, aulões e tudo que for rastreável ficarão disponível pra consulta do usuário sendo do dia 0 que ele entrou na comunidade até o 31 de dezembro.

## Dúvidas

* Devemos resetar o /apresentar e só iniciar a temporada para pessoas que executarem o comando novamente? 
* O que podemos dar para pessoas que acabaram no topo do Ranking?
* Deveriamos começar a pedir a vinculação de e-mail da rapaziada pra mail marketing da He4rt? Pois já pode ser um momento propício. Ex: aulões, eventos etc.

**Comment by gitlherme** [comment]
Acredito que resetar temporada também é uma boa ideia pra ter entendimento de como está o engajamento da comunidade.
Acho que resetar o /apresentar, nesse primeiro momento talvez faça sentido, mas não se tornar uma coisa que deva fazer todo ano.

Sobre premiação, acho que pode ser coisas que ajudem a pessoa a se desenvolver na área. Seja sendo um curso x que ela queira, podendo ser num valor mais caro, sem ser cursos da Udemy, ou periféricos que possam ajudar no dia a dia, como teclados melhores, mouses, microfones, webcams ou até cadeiras.

Acredito que talvez possa encontrar uma forma de fazer alguma integração do Discord com a plataforma Meetup que tem sido usada, assim vc não precisa pedir o e-mail para pessoas que entrem no Discord, mas sim para aquelas que façam parte do grupo no Meetup, centralizando mais para esses eventos como aulões, meetups, etc. Caso isso não seja possível, acredito que e-mail possa ser um dos dados do comando /apresentar, porém deixando-o oculto no perfil do usuário, tendo isso só pra moderação mesmo.

**Comment by diogoodev** [comment]
Acredito que poderia ter um espaço para exibir todos os selos já conquistados, assim mesmo resetando a temporada ainda seria possível ver como foram as anteriores daquele usuário. Ai faria sentido resetar para todo mundo a cada nova temporada.

> Acredito que resetar temporada também é uma boa ideia pra ter entendimento de como está o engajamento da comunidade. Acho que resetar o /apresentar, nesse primeiro momento talvez faça sentido, mas não se tornar uma coisa que deva fazer todo ano.
>

**Comment by gitlherme** [comment]
Concordo, @diogoodev. Acho só que talvez a pessoa não precisasse se apresentar todo ano, saca? Acho que o ideal é ela poder ir alterando seu perfil, mas sem ter que se apresentar sempre. Talvez entender um outro jeito de resetar e colocar a pessoa na "competição" novamente, mas sem precisar rodar o /apresentar.

**Comment by yayaflc** [comment]
Acho válido resetar o apresentar e iniciar somente com aqueles que executarem o comando, tendo em vista que durante esses 4 anos muitas pessoas abandonaram o barco. 

Para o top ranking seria interessante uma badge especial e um mimo irl como honorária (quem sabe um chaveiro 3D edição especial 👀).

A vinculação de email acredito que seja meio delicado, já que dentro do servidor temos muitos eventos onlines semanais (aulões/reuniões). Acabaria virando muito spam em uma caixa de entrada, já que email geralmente é utilizado para assuntos importantes, pessoais e de trabalho. 
O bot da He4rt enviando uma mensagem direta no Discord, alertando sobre os eventos é uma possibilidade?

**Comment by Luisnadachi** [comment]
Sobre mostrar a temporada anterior, pode utilizar algo parecido que tem no [op gg](https://www.op.gg/summoners/br/Nadachi), principalmente em questão de tamanho pra não ocupar tanto espaço. Sobre as dúvidas: 

- Não acho certo resetar o apresentar por conta dos futuros impactos que isso pode causar no futuro: membros perdidos depois de muito tempo fora, números baixos, privar de algumas coisas.
- Acho que talvez algum periférico ou algo relacionado a tecnologia.
- Acho válido pedir o email mas não ser algo obrigatório agora de inicio.
- Justo dar a conquista depois de um certo level para evitar pessoas digitando somente uma vez para conquista-la e limitando o raking para pessoas que realmente estão engajando e participando.

**Comment by diogoodev** [comment]
> Sobre mostrar a temporada anterior, pode utilizar algo parecido que tem no [op gg](https://www.op.gg/summoners/br/Nadachi), principalmente em questão de tamanho pra não ocupar tanto espaço

Isso é muito bom. Tenho uma sugestão para complementar com o sistema de avaliações, ele poderia usar uma métrica de estrelas para os voluntários. Assim iria gerar um diferencial no perfil da pessoa voluntaria, fiz um rascunho no canva só pra demonstrar a ideia. 
![Melhor temporada (11)](https://user-images.githubusercontent.com/20946986/207213035-ab1aa85e-d45b-4b69-b90a-7208ef595e84.png)
